### PR TITLE
Part of issue 1101 and case insensitivity and tests.

### PR DESCRIPTION
**Pull request by evilebottnawi**
/cc @davidtheclark

### CODE REVIEW

File: src/utils/isCustomIdentPropertyAnimationName.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * @return {boolean} If `true`, value is a custom ident
+ */
+
+// https://developer.mozilla.org/ru/docs/Web/CSS/animation-name
+const knownValues = new Set([
+  "none", "inherit", "initial", "unset",
+])
+
+export default function (value) {
+  if (knownValues.has(value.toLowerCase())) { return false }
```
Comment (davidtheclark) [inline src/utils/isCustomIdentPropertyAnimationName.js]: You can shorten this by just doing `return knownValues.has(value.toLowerCase())`, right?

**Comment by davidtheclark** [comment]
We have two different kinds of `isSomething` functions going. One is just a simple check to see if a word is in a Set — that's like the ones you're adding here. Most of them so far, though, are not a little less straightforward (e.g. the `isStandardSomething` functions).

I kind of wonder if instead of creating new utilities for the Set checkers (with tests), we just create a new directory, like `keywordSets`, and put all our sets of keywords in there, and the import and use those as needed.

The idea is that the `isSomething` functions would always be slightly more complicated than just checking a Sets. And the Sets, which are just the product of research and don't really need tests, live together in a distinct directory.

What do you think?

This would address https://github.com/stylelint/stylelint/issues/1071, too.

**Comment by evilebottnawi** [comment]
@davidtheclark i think it is good idea, i suggest create multiple utils for resolve issue as quickly as possible, then refactoring in one function

**Comment by davidtheclark** [comment]
I'm not suggesting refactoring into a single function, though, but creating a directory of Sets. I think that if we want that approach, we should start it right away, then gradually move things into it.

**Comment by evilebottnawi** [comment]
@davidtheclark I think we have to start doing it after `7.0.0`, we have many bugs, which we should resolve. Also, I think you need a good design it.

**Comment by davidtheclark** [comment]
I'm not sure it's at all as complicated as you seem to think it is. It's no
harder than writing more util functions. I will get a start on it sometime
today.

**Comment by davidtheclark** [comment]
Feel free to merge after green light.

**Comment by evilebottnawi** [comment]
@davidtheclark good, I was suspended for today and tomorrow start to finish through your proposal

**Comment by evilebottnawi** [comment]
CI green. Merge.

**Pull request merged**